flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobInfo.scala

```diff
@@ -37,8 +37,10 @@ import org.apache.flink.runtime.akka.ListeningBehaviour
  * @param start Starting time
  */
 class JobInfo(
-  val client: ActorRef,
-  val listeningBehaviour: ListeningBehaviour,
+
+  var client: ActorRef,
+  var listeningBehaviour: ListeningBehaviour,
```

Comment:
We might want to allow multiple clients here. Otherwise only the least recent registered client will receive updates.